### PR TITLE
ItemEntity: Add message to moveresult assertion

### PR DESCRIPTION
**Pull request by rubenwardy**
See #9796

### CODE REVIEW

**Comment by rubenwardy** [comment]
Better message needed, one that preferably doesn't need to be split (the concat will run even if the assert doesn't fail)

**Comment by sfan5** [comment]
`"collision info missing, this is caused by a out-of-date mod or game"`
short enough?

File: builtin/game/item_entity.lua [hunk 1]
```diff
@@ -200,7 +200,11 @@ core.register_entity(":__builtin:item", {
 			return -- Don't do anything
 		end
 
-		assert(moveresult)
+		assert(moveresult,
+				"Collision info not passed to Item Entity's on_step. " ..
```
Comment (rubenwardy) [inline builtin/game/item_entity.lua:204]:
```suggestion
			"Collision info missing, this is caused by an out-of-date/buggy mod or game")
```

File: builtin/game/item_entity.lua [hunk 2]
```diff
@@ -200,7 +200,11 @@ core.register_entity(":__builtin:item", {
...
+				"This is caused by a game (ie: Minetest Game) or mod not overriding correctly, make sure to update them.")
```
Comment (rubenwardy) [inline builtin/game/item_entity.lua:205]:
```suggestion
```

File: builtin/game/item_entity.lua [hunk 3]
```diff
@@ -200,7 +200,10 @@ core.register_entity(":__builtin:item", {
 			return -- Don't do anything
 		end
 
-		assert(moveresult)
+		assert(moveresult,
+			"Collision info missing, this is caused by an out-of-date/buggy mod or game")
+
+
```
Comment (rubenwardy) [inline builtin/game/item_entity.lua:206]:
```suggestion
```

**Comment by SmallJoker** [comment]
Could you please do the same for the falling node?
Take my :+1: when done.

File: builtin/game/item_entity.lua [hunk 4]
```diff
@@ -200,7 +200,9 @@ core.register_entity(":__builtin:item", {
 			return -- Don't do anything
 		end
 
-		assert(moveresult)
+		assert(moveresult,
+			"Collision info missing, this is caused by an out-of-date/buggy mod or game")
```
Comment (sfan5) [inline builtin/game/item_entity.lua:204]:
```suggestion
			"Collision info missing, this is caused by an outdated/buggy mod or game")
```

Comment (sfan5) [reply builtin/game/item_entity.lua:204]: nevermind actually

**Pull request merged**